### DISCUSSION TITLE
Running Watchtower to automatically update container base images?

### DISCUSSION THREAD

**Discussion by louwers** [dokku/dokku#4586] [Q&A] [answered]
Hello.

I was just wondering if running [Watchtower](https://containrrr.dev/watchtower/) along with Dokku would work to keep Docker base images up to date. If not, does Dokku contain a similar mechanism to automatically re-deploy applications whose Docker container base image has been updated?

**Comment by Cellane** [comment, marked as answer]
Hello, I think I can tackle this one!

So when I was researching Watchtower a long while back (meaning, something might have changed since then), it worked by pulling a new image from Docker registry and restarting the running container with a new image. Assuming you’re deploying your Docker images with Dokku’s `git:from-image` command (or formerly `tags:deploy`), that’s not what you want to be doing – you would want upon detection of an updated upstream image to run an appropriate Dokku command to deploy that image, but alas, I don’t think Watchtower can run a custom command once it detects an image.

That leaves you with two options that I’m aware of.

The first one would be to use a different utility that monitors upstream registries and notifies you of updated images but allows you to specify a custom command to run upon detection of an updated image. One such tool would be [Diun](https://crazymax.dev/diun/), a tool I’m happily using myself to pull all updated upstream images of Bitwarden, adjust them just a little bit for better Dokku compatibility, build and push the updated image to my personal Docker registry. The initial setup is a bit daunting but once configured, it works really well.

The second one would be my tool Shinka which is specifically written to allow you to seamlessly update your Dokku apps with updated Docker images, but here’s a problem – there are two versions of Shinka right now and neither is really ready to be deployed without you having to do a bit of manual work. Former [Shinka](https://github.com/Cellane/shinka) was not updated in quite a while and I wonder if it will work properly with recent versions of Dokku. [Shinka Next](https://github.com/Cellane/shinka-next) definitely works with the most recent version of Dokku, but I haven’t had the time to write proper documentation and installation/deployment instructions just yet (or even think about those, for that matter). If that’s not something that would discourage you, feel free to hop on Dokku’s Slack and I’ll do my best to assist you 👌
